File: Daily Test/C++/Submatrix Increment Pattern 30.8.2025.cpp
#include <iostream>
using namespace std;

int main() {
    int M, N;
    cin >> M >> N;
    int start = 1, copy = 1;
    int size = M * N;
    int mat[size][size];

    for (int i = 0; i < size; i += M) {
        for (int j = 0; j < size; j += M) {
            for (int k = i; k < i + M; k++) {
                for (int l = j; l < j + M; l++) {
                    mat[k][l] = start++;
                }
            }
            start = copy + 1;
            copy++;
        }
    }

    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {
            cout << mat[i][j] << " ";
        }
        cout << endl;
    }
    return 0;
}

File: Daily Test/Python/Submatrix Increment Pattern 30.8.2025.py
M, N = map(int, input().split())
start = 1
copy = 1
mat = [[0 for i in range(M * N)] for j in range(M * N)]

for i in range(0, M * N, M):
    for j in range(0, M * N, M):
        for k in range(i, i + M):
            for l in range(j, j + M):
                mat[k][l] = start
                start += 1
        start = copy + 1
        copy += 1

for i in mat:
    print(*i)